### ISSUE TITLE
Alerting: Reloading provisioned alerts resets all alerts states

### ISSUE THREAD

**Issue by azelezni** [grafana/grafana#61917]
Hello, we started using Grafana unified Alerts with file provisioning, and noticed some alerts have their state reset at seemingly random times.
After further investigation it seems that any call to `/api/admin/provisioning/alerting/reload` will:
1. Recreate *all* alerts (you can track this in `alert_rule_version` table)
2. Reset state of all alerts to Normal

Now, I understand why an alert state would be reset if I changed something in the alert config, but in this case **nothing** was changed, I simply POST to `/api/admin/provisioning/alerting/reload` and all my alerts are reset, so now I get duplicate notifications that just confuse people.

I tested with alerts created from the UI and this doesn't happen, if I save an alert without any changes its state remains the same, it only resets when a change is made, which makes sense.

That reset functionality **should not** reset all the alerts if nothing changed, this just causes a lot of noise.
Basically this functionality makes alerts & file-provisioning useless since I can't know if an alert was fired/closed because Grafana was reloaded or an actual issue is happening.

This is also related https://github.com/grafana/grafana/issues/53302, in this case no alert could ever be fired, since every minutes it's state is reset.

This exact same issue happened in Prometheus a few years ago and was fixed by simply keeping the state of the alert during reload.
https://github.com/prometheus/prometheus/issues/3376
https://github.com/prometheus/prometheus/pull/3382

**Comment by olivierperdaens** [comment]
For me, the same happens with alerts created through the GUI: 

If I edit an alert of let's say folder A, group AA, then all my alerts from the same folder>group are reset ... Expected behavior: only changed alert is reset !

**Comment by yuri-tceretian** [comment]
PR https://github.com/grafana/grafana/pull/66531 should fix this too.

**Comment by yuri-tceretian** [comment]
Closing via https://github.com/grafana/grafana/pull/66531

**Comment by yuri-tceretian** [comment]
backported to 9.5.x, and will be released in 9.5.2

**Comment by andykingit** [comment]
I dont think it was released in 9.5.2? according to the [CHANGELOG](https://raw.githubusercontent.com/grafana/grafana/main/CHANGELOG.md) for 9.5.2

**Comment by yuri-tceretian** [comment]
I double checked: the commit is in the branch release-9.5.2, and scheduler logs "Alert rule evaluated" contain the new context `fingerprint`

**Comment by andykingit** [comment]
Thanks, and I tested it out and all works out nicely, I guess the backporting does not include it in the changelog
